### PR TITLE
test: backend e2e test for route with jwt (no algorithm)

### LINKED ISSUE #1190: bug: when enable jwt-auth plugin on dashabord, can't get the token of jwt-auth
# Bug report

when enable jwt-auth plugin on dashabord, can't get the token of jwt-auth

## How to Reproduce

1. create consumer with jwt-auth
```
Request URL: http://127.0.0.1:9000/apisix/admin/consumers
Request Method: PUT
Request Body: {"username":"test_jwt","plugins":{"jwt-auth":{"disable":false,"exp":86400,"key":"user-key","secret":"my-secret-key"}}}
Response: {"code":0,"message":"","data":null,"request_id":"07e5ef1a-061c-4649-a834-b6bff8b29440"}
```
2. get the token of jwt-auth
```
$ curl http://127.0.0.1:9080/apisix/plugin/jwt/sign?key=user-key -i
HTTP/1.1 500 Internal Server Error
Date: Mon, 04 Jan 2021 06:10:50 GMT
Content-Type: text/plain; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Server: APISIX/2.1

failed to sign jwt
```

### PR DESCRIPTION
**Pull request by idbeta**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] E2E test
- [ ] New feature provided
- [ ] Improve performance

close #1190

### CODE REVIEW

File: api/test/e2e/consumer_test.go [hunk 1]
```diff
@@ -536,3 +536,70 @@ func TestConsumer_with_createtime_updatetime(t *testing.T) {
...
+
+		{
+			Desc:         "get the token of jwt ",
+			Object:       APISIXExpect(t),
+			Path:         "/apisix/plugin/jwt/sign",
+			Query:        "key=user-key",
+			Method:       http.MethodGet,
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
```
Comment (nic-chen) [inline api/test/e2e/consumer_test.go:590]: need to valid the token on DP.

Comment (idbeta) [reply api/test/e2e/consumer_test.go:622]: fixed.

File: api/test/e2e/consumer_test.go [hunk 2]
```diff
@@ -536,3 +536,119 @@ func TestConsumer_with_createtime_updatetime(t *testing.T) {
...
+		{
+			Desc:         "hit route with jwt token",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"Authorization": string(respBody)},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+			Sleep:        sleepTime,
+		},
```
Comment (nic-chen) [inline api/test/e2e/consumer_test.go:622]: Since the name is test plugin jwt, we should add test cases to test scenarios without token and wrong token.

Comment (idbeta) [reply api/test/e2e/consumer_test.go]: I found that the case `route_with_auth_plugin_test.go` already contains what you said, I added this new scenario here, so it should be easier to understand.

**Comment by idbeta** [comment]
ping @membphis

File: api/test/e2e/route_with_auth_plugin_test.go [hunk 3]
```diff
@@ -17,14 +17,16 @@
...
+	"io/ioutil"
 	"net/http"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestRoute_With_Auth_Plugin(t *testing.T) {
+func TestRoute_With_Jwt_Plugin(t *testing.T) {
```
Comment (membphis) [inline api/test/e2e/route_with_auth_plugin_test.go:29]: the file name should contain `jwt` too

File: api/test/e2e/route_with_auth_plugin_test.go [hunk 4]
```diff
@@ -173,4 +175,118 @@ func TestRoute_With_Auth_Plugin(t *testing.T) {
...
+	basepath := "http://127.0.0.1:9080"
+	request, _ := http.NewRequest("GET", basepath+"/apisix/plugin/jwt/sign?key=user-key", nil)
+	request.Header.Add("Authorization", token)
+	resp, err := http.DefaultClient.Do(request)
+	if err != nil {
+		fmt.Printf("server not responding %s", err.Error())
+	}
+	defer resp.Body.Close()
+	assert.Equal(t, 200, resp.StatusCode)
+	respBody, _ := ioutil.ReadAll(resp.Body)
```
Comment (membphis) [inline api/test/e2e/route_with_auth_plugin_test.go:247]: need a better name for `respBody`

File: api/test/e2e/route_with_auth_plugin_test.go (hunk 3 above)
Comment (idbeta) [reply api/test/e2e/route_with_auth_plugin_test.go:29]: The current name is convenient for adding other auth plugin cases here in the future.

File: api/test/e2e/route_with_auth_plugin_test.go (hunk 4 above)
Comment (idbeta) [reply api/test/e2e/route_with_auth_plugin_test.go]: fixed.

File: api/test/e2e/route_with_auth_plugin_test.go [hunk 5]
```diff
@@ -173,4 +175,118 @@ func TestRoute_With_Auth_Plugin(t *testing.T) {
...
+		testCaseCheck(tc, t)
+	}
+
+	// get the token of jwt
+	basepath := "http://127.0.0.1:9080"
+	request, _ := http.NewRequest("GET", basepath+"/apisix/plugin/jwt/sign?key=user-key", nil)
+	request.Header.Add("Authorization", token)
+	resp, err := http.DefaultClient.Do(request)
+	if err != nil {
+		fmt.Printf("server not responding %s", err.Error())
```
Comment (starsz) [inline api/test/e2e/route_with_auth_plugin_test.go:243]: It looks not good. Maybe assert.Nil(t, err).

Comment (idbeta) [reply api/test/e2e/route_with_auth_plugin_test.go]: good point, fixed.

File: api/test/e2e/route_with_auth_plugin_test.go (hunk 3 above)
Comment (membphis) [reply api/test/e2e/route_with_auth_plugin_test.go:28]: and I think we should update the file name too

File: api/test/e2e/route_with_auth_plugin_test.go [hunk 6]
```diff
@@ -173,4 +174,116 @@ func TestRoute_With_Auth_Plugin(t *testing.T) {
...
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+	}
+
+	for _, tc := range tests {
+		testCaseCheck(tc, t)
+	}
+
```
Comment (membphis) [inline api/test/e2e/route_with_auth_plugin_test.go:288]: code style: remove this line

**Comment by nic-chen** [comment]
@idbeta  please resolve the reviews.

File: api/test/e2e/route_with_auth_plugin_test.go (hunk 6 above)
Comment (idbeta) [reply api/test/e2e/route_with_auth_plugin_test.go]: fixed.

File: api/test/e2e/route_with_auth_plugin_test.go (hunk 3 above)
Comment (idbeta) [reply api/test/e2e/route_with_auth_plugin_test.go:28]: fixed.

File: api/test/e2e/route_with_plugin_jwt_test.go [hunk 7]
```diff
@@ -173,4 +174,115 @@ func TestRoute_With_Auth_Plugin(t *testing.T) {
...
+			Desc:         "delete consumer",
+			Object:       ManagerApiExpect(t),
+			Path:         "/apisix/admin/consumers/consumer_1",
+			Method:       http.MethodDelete,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "\"code\":0",
+		},
+		{
+			Desc:         "after delete consumer verify it again",
```
Comment (nic-chen) [inline api/test/e2e/route_with_plugin_jwt_test.go:267]: we could verify it by hitting the route too.

**Review by membphis — APPROVED** [review]

File: api/test/e2e/route_with_plugin_jwt_test.go (hunk 7 above)
Comment (juzhiyuan) [reply api/test/e2e/route_with_plugin_jwt_test.go:267]: @nic-chen 

image

Does this need to be updated?

Comment (nic-chen) [reply api/test/e2e/route_with_plugin_jwt_test.go:267]: why marked this conversation as resolved? If you think it's unnecessary, just leave your comment.

Comment (idbeta) [reply api/test/e2e/route_with_plugin_jwt_test.go:267]: I think it's unnecessary.

**Review by nic-chen — APPROVED** [review]

**Pull request merged**